zrml/swaps/src/tests.rs

```diff
@@ -3506,10 +3712,425 @@ fn execute_arbitrage_observes_min_balances_mint_sell() {
+
+        assert_err!(
+            Swaps::swap_exact_amount_in(
+                alice_signed(),
+                DEFAULT_POOL_ID,
+                asset_in,
+                asset_amount_in,
+                asset_out,
+                None,
+                Some(max_spot_price),
+            ),
+            Error::<Runtime>::BadLimitPrice
+        );
+    });
+}
+
+#[test_case(BASE_ASSET, ASSET_B; "base_asset_in")]
+#[test_case(ASSET_B, BASE_ASSET; "base_asset_out")]
+#[test_case(ASSET_B, ASSET_C; "no_base_asset")]
+fn swap_exact_amount_out_creator_fee_charged_correctly(
+    asset_in: Asset<MarketId>,
+    asset_out: Asset<MarketId>,
+) {
+    ExtBuilder::default().build().execute_with(|| {
+        let creator_fee = Perbill::from_percent(1);
+        let swap_fee = 0;
+
+        assert_ok!(set_creator_fee(DEFAULT_MARKET_ID, creator_fee));
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, Some(swap_fee), true);
+
+        let market_creator = MarketCommons::market(&DEFAULT_MARKET_ID).unwrap().creator;
+        let market_creator_balance_before = Currencies::free_balance(BASE_ASSET, &market_creator);
+        let pool_account = Swaps::pool_account_id(&DEFAULT_POOL_ID);
+        let pool_balance_in_before = Currencies::free_balance(asset_in, &pool_account);
+        let pool_balance_out_before = Currencies::free_balance(asset_out, &pool_account);
+        let pool_balance_base_out_before = Currencies::free_balance(BASE_ASSET, &pool_account);
+        let asset_amount_out = _1;
+        let max_asset_amount_in = Some(u128::MAX);
+        let max_price = None;
+        let expected_in_without_fee = calc_in_given_out(
+            pool_balance_in_before,
+            DEFAULT_WEIGHT,
+            pool_balance_out_before,
+            DEFAULT_WEIGHT,
+            asset_amount_out,
+            swap_fee,
+        )
+        .unwrap();
+
+        assert_ok!(Swaps::swap_exact_amount_out(
+            alice_signed(),
+            DEFAULT_POOL_ID,
+            asset_in,
+            max_asset_amount_in,
+            asset_out,
+            asset_amount_out,
+            max_price,
+        ));
+
+        let market_creator_balance_after = Currencies::free_balance(BASE_ASSET, &market_creator);
+
+        let expected_fee = if asset_in == BASE_ASSET {
+            creator_fee.mul_floor(expected_in_without_fee)
+        } else if asset_out == BASE_ASSET {
+            creator_fee.mul_floor(asset_amount_out)
+        } else {
+            let fee_before_swap = creator_fee.mul_floor(asset_amount_out);
+            calc_out_given_in(
+                DEFAULT_LIQUIDITY - asset_amount_out - fee_before_swap,
+                DEFAULT_WEIGHT,
+                pool_balance_base_out_before,
+                DEFAULT_WEIGHT,
+                fee_before_swap,
+                swap_fee,
+            )
+            .unwrap()
+        };
+
+        assert_eq!(market_creator_balance_after - market_creator_balance_before, expected_fee);
```

Comment:
I disagree that those checks are necessary, because:
1. The actual swap logic has not been changed in this PR. Fees are only deducted before the swap or after the swap, the swap logic remained the same though.
Consequently the check "Verify that the pool's balances have changed as expected." is not necessary.
2. In addition to 1., in the case of 
a. `swap-exact-amount-in`: The fees that are charged are tested already. The swap logic remained untouched. Another test already verifies that `min_asset_amount_out` is respected properly in the case of creator fee. Thus, payout of desired amount is checked.
b. `swap-exact-amount-out`: The fees that are charged are tested already. The swap logic remained untouched. In addition to that, the `asset_amount_out` is verified and the `max_asset_amount_in` is verified, thus ensuring that the swap happens according to the users configuration.

Consequently, unless convinced otherwise, I'll mark those as "won't do".

Reply:
Yeah, that's fine. I agree with the perspective that we're already testing this somewhere. The only counterargument is that our tests suite is pretty bloated and it's sometimes hard to tell what we might be losing track of.